i2cEncoder.py

```diff
@@ -53,45 +79,45 @@ def change_color(encoder):
 
     # Set color Blue
     if counter < MAX_VAL / 2:
-        colorStep = counter + 1 ** EXP_GROWTH 
+        colorStep = counter + 1 ** EXP_GROWTH
         color = '0x0000{:02x}'.format(int(255 / colorStep))
         printColor = "Blue"
 
     if (not encoder.readStatus(i2cEncoderLibV2.RMAX) and
             not encoder.readStatus(i2cEncoderLibV2.RMIN)):
         encoder.writeRGBCode(int(color, 0))
-        print('Counter: {} {}: {}'.format(counter, printColor, color))
+        send_msg(osc_address_patterns[idx], channel, counter)
+        print('{} {} {} ({}, {})'.format(osc_address_patterns[idx], channel, counter, printColor, color))
 
 
 encoders = []
 for idx, channel in enumerate(DEVICES):
     encoders.append([])
     for value in channel:
-        print(value)
         encoders[idx].append(init_encoder(value))
 
 try:
-    for channel, row in enumerate(cycle(encoders)):
-        for encoder in row:
+    for channel, row in cycle(enumerate(encoders)):
+        for idx, encoder in enumerate(row):
```

Comment:
Aquest `idx` declara aqui no s'usa.
El vols usar a la linia 89, pero es un altra funcio,alla no es visible

Reply:
Si el vols usar l'hauras de passar a `run_encoder`:
`run_encoder(encoder, idx)`

I despres  a la 67:
```
def run_encoder(encoder, idx):
```

Reply:
Pero to plegat es redundant, pq estas lligant els DEVICES a encoders:
```
DEVICES = [
    [0x21, 0x5, 0x41, 0x9, 0x40],
    [0x11, 0x20, 0x3],
]
```
I despres declares els 
```
osc_address_patterns = ["/switch", "/amp", "/rate", "/start", "/length"]
osc_address_patterns_alt = ["/attach", "/pitch"]
```

Per acabar lligant ->
```
0x21 --> "/switch"
0x5 -->  "/amp"
# etc
```

Seria mes facil lligar-los al primer moment:
```
DEVICES = [
    [(0x21, "/switch"), (0x5, "/amp)"...
]
```

Reply:
> run_encoder

Si que faig servir `idx` a la línia 89. I funciona sense passar-li l'argument a `run_encoder`.

Reply:
Això no és òptim:

DEVICES = [
    [(0x21, "/switch"), (0x5, "/amp)"...
]

perquè pensa que no està acabat i que acabaria sent així

DEVICES = [
    [(0x21, "/switch"), (0x5, "/amp)"...
    [(0x11, "/switch"), (0x20, "/amp)"...
    [(0xXX, "/switch"), (0xXX, "/amp)"...
    [(0xXX, "/switch"), (0xXX, "/amp)"...
    [(0xXX, "/switch"), (0xXX, "/amp)"...
]

I que a més a més els encoders envien un missatge o altre segons si tenim pulsat el butó Alt, el qual encara no tinc implementat perquè físicament encara no el tinc. Aquest butó Alt és com si fos un Shift, i llavors en comptes d'enviar els patterns /switch, /amp... enviarà /attach, /pitch...

Reply:
> > run_encoder
> 
> Si que faig servir `idx` a la línia 89. I funciona sense passar-li l'argument a `run_encoder`.

No esta declara a dins de `run_encoder`, no te definit vaor, no ot funcionar

Reply:
> Això no és òptim:
> 
> DEVICES = [
> [(0x21, "/switch"), (0x5, "/amp)"...
> ]
> 
> perquè pensa que no està acabat i que acabaria sent així
> 
> DEVICES = [
> [(0x21, "/switch"), (0x5, "/amp)"...
> [(0x11, "/switch"), (0x20, "/amp)"...
> [(0xXX, "/switch"), (0xXX, "/amp)"...
> [(0xXX, "/switch"), (0xXX, "/amp)"...
> [(0xXX, "/switch"), (0xXX, "/amp)"...
> ]
> 
> I que a més a més els encoders envien un missatge o altre segons si tenim pulsat el butó Alt, el qual encara no tinc implementat perquè físicament encara no el tinc. Aquest butó Alt és com si fos un Shift, i llavors en comptes d'enviar els patterns /switch, /amp... enviarà /attach, /pitch...

Llavors pensa en quina sera la *estructura final* del model de dades, pq si no, si vas fent el programa incrementalment obigues a canviar les funcions cada cop que hi ha un canvi en la estructura de dades, en aquest cas DEVICES i els patterns.

O sigui, tot girava abans en funcio de DEVICES, i les funcions iels loops tenien en compte aixo, si ara inrodueixes els patterns s'ha de fer lloc a funcions o loops que no ho tenien previst, i llavors es complica mol tot.

Si la esructura final es com la que dius, fes-la ja, encara que el 90% de les dades estiguin buides,  llavors es poden fer les ffuncions i els loops ja pensant en aixo